sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -39,6 +40,19 @@ namespace Azure { namespace Core {
      */
     std::string ToString();
 
+    /**
+     * @brief Construct a Uuid from an existing UUID represented as an array of bytes.
+     * @details Creates a Uuid from a UUID created in an external scope.
+     */
+    explicit Uuid(std::array<uint8_t, UuidSize> const& uuid) { m_uuid = uuid; }
```

Comment:
I am a bit weary of allowing arbitrary bytes to be interpreted as a uuid without validation.

We should consider verifying this is well-formed on construction.

Reply:
Can you share the sample code paths in AMQP that will call this, for interopability?

Reply:
Also, add changelog entry.

Reply:
A UUID is by definition 128 bits of arbitrary data. I'm not sure what "validation" can be done here. 

Could you please give some examples of the validations you are contemplating?

Reply:
Here's some sample code of how this is used (from the unit tests):
```cpp
TEST(Uuid, Transparent)
{
  auto uuid1 = Uuid::CreateUuid();
  auto arrayUuid1(uuid1.GetAsArray());
  auto uuid2 = Azure::Core::Uuid(arrayUuid1);
  EXPECT_EQ(uuid1.ToString(), uuid2.ToString());
}
```

Reply:
> A UUID is by definition 128 bits of arbitrary data. I'm not sure what "validation" can be done here.
> 
> Could you please give some examples of the validations you are contemplating?

Good point. I was initially thinking about string/character validation which doesn't apply here.

That said, we may want to reject things that are reserved. For example, there are a set of bytes we won't create:
https://github.com/Azure/azure-sdk-for-cpp/blob/d60a995bb47b738672ad00080ff215fe66a309e8/sdk/core/azure-core/src/uuid.cpp#L70-L71

Reply:
I am not at all comfortable with adding this kind of validation, especially since we don't do any other similar validations of values we accept from customers (for instance, if we accept strings that should be interpreted as UTF-8, we don't validate the surrogate pairs), and (to my knowledge), no other code which accepts constructs which are ostensibly UUIDs validate those bits.